### DISCUSSION TITLE
drop C requirement

### DISCUSSION THREAD

**Discussion by 3052** [typst/typst#3590] [General]
I want to built Typst so I can try out
https://github.com/typst/typst/pull/3571

but I dont have a C compiler (on principle). so currently I cannot build Typst:

~~~
warning: psm@0.1.21: Failed to run: "gcc.exe" "--version"
error: failed to run custom build command for `psm v0.1.21`
~~~

is it possible for Typst to drop this and any other C crates, so that I can build with Rust only?

**Comment by frogtd** [comment]
If you create a .cargo/config.toml file with contents such as:
```toml
[env]
CARGO_CFG_TARGET_ARCH = "blah"
```
that dependency will simply do nothing and compile without a C compiler.

**Reply by 3052** [reply]
nope, that doesn't work. if I create `~/.cargo/config.toml` with:

~~~toml
[env]
CARGO_CFG_TARGET_ARCH = "psm"
~~~

I get the exact same error above. same error with this too:

~~~toml
[env]
CARGO_CFG_TARGET_ARCH = "x86_64"
~~~

**Reply by frogtd** [reply]
I did not have a ~/?

**Reply by 3052** [reply]
same failure with `.cargo/config.toml`. maybe you should actually test your advice before sending people on wild goose chases.

**Reply by frogtd** [reply]
I did

**Reply by 3052** [reply]
OK, well then provide the exact file content that you used.

**Reply by frogtd** [reply]
I did

**Reply by 3052** [reply]
nope, you are wrong:

~~~
> Get-Content .\.cargo\config.toml
[env]
CARGO_CFG_TARGET_ARCH = "blah"

> $env:path = 'D:\Rust\bin'

> cargo build -p typst-cli --no-default-features
warning: psm@0.1.21: Failed to run: "gcc.exe" "--version"
error: failed to run custom build command for `psm v0.1.21`
~~~
